test/entice/logic/casting_test.exs

```diff
@@ -0,0 +1,73 @@
+defmodule Entice.Logic.CastingTest do
+  use ExUnit.Case, async: true
+  alias Entice.Entity
+  alias Entice.Entity.Test.Spy
+  alias Entice.Logic.Skills
+  alias Entice.Logic.Casting
+  alias Entice.Logic.Vitals.Energy
+
+
+  setup do
+    {:ok, entity_id, _pid} = Entity.start
+    Casting.register(entity_id)
+    Spy.register(entity_id, self)
+    Entity.put_attribute(entity_id, %Energy{mana: 50})
+    {:ok, [entity_id: entity_id]}
+  end
+
+  @tag id: 0, casting: true
```

Comment:
id we should use for setup but Im too lazy to do it. 
casting I use to do mix test --only casting it's faster for repeat tests. We could do this via file path but it's cleaner that way.